core/chains/evm/config/config.go

```diff
@@ -49,6 +49,7 @@ type EVM interface {
 	NodeNoNewHeadsThreshold() time.Duration
 	FinalizedBlockOffset() uint32
 	NoNewFinalizedHeadsThreshold() time.Duration
+	TXMEnabled() bool
```

Comment:
We already have `EVM.Transactions`. Would `EVM.Transactions.XYZ` make sense? Rather than `EVM.TXMEnabled`?
- `EVM.Transactions.Enabled`
- `EVM.Transactions.ManagerEnabled`
- ??

Reply:
I considered adding it there but since this also affects the gas estimator I thought it was better to keep the config at the EVM level like `LogBroadcasterEnabled`. I'm not steadfast on that though so not against moving this under `EVM.Transactions`. That does remind me that I should update the config description to mention the gas estimator though.

Reply:
Why does the gas estimator change things? Does that suggest that we might have wanted it nested as `Transactions.GasEstimator`?

Reply:
I think it still makes sense to have gas estimator configs outside of `EVM.Transactions` since that component could still be referenced separately. But in the case for chain initialization, the gas estimator depends on the TXM to [start its lifecycle](https://github.com/smartcontractkit/chainlink/blob/ed6f486d59c2cbdfa02c6fbfc3a42fb6d5f805a2/core/chains/legacyevm/chain.go#L247) which is why they go hand in hand.

Reply:
OK, but I am still not seeing how this new field is relevant to the gas estimator. The `TXMEnabled()` method is only called from one place currently.

Reply:
The only relevance is that the new field is also responsible for enabling/disabling gas estimators along with the TXM. Maybe the config name is misleading since it only refers to the TXM.

Reply:
I think it should at least be expanded for clarity, but if there is a more general name that would be good.
```suggestion
	TransactionManagerEnabled() bool
```

Reply:
Updated the name in the latest commit. I couldn't think of a more general name without it becoming too verbose. I've mentioned how this affects gas estimators in the config description though.

Reply:
With the gas estimator decoupled from this setting, should we reconsider something like `Transactions.Enabled()`?

Reply:
So it's not completely decoupled since we still rely on the TXM to start/stop the gas estimator. But I was curious what @dimriou's opinion would be for the scope of this config though just to get a second opinion.